### ISSUE TITLE
ScaleBar and DisappearingScaleBar cause crash on first composition for failed resolution

### ISSUE THREAD

**Issue by FilippoVigani** [googlemaps/android-maps-compose#246]
#### Steps to reproduce

  1. Use the following imports:
  
```
    implementation("com.google.android.gms:play-services-maps:18.1.0")

    implementation("com.google.maps.android:maps-compose:2.8.0")

    implementation("com.google.maps.android:maps-compose-widgets:2.8.0")
```
  3. Add a ScaleBar composable
  4. Run the application

#### Code example

```
Box(modifier = Modifier
                .sizeIn(maxWidth = 400.dp)
                .aspectRatio(1f)
                .fillMaxWidth()) {
                GoogleMap(
                    modifier = Modifier
                        .matchParentSize(),
                    locationSource = locationSource,
                    properties = mapProperties,
                    cameraPositionState = cameraPositionState,
                    uiSettings = mapUiSettings,
                )
                DisappearingScaleBar(
                    modifier = Modifier
                        .padding(top = 8.dp)
                        .align(Alignment.TopStart),
                    cameraPositionState = cameraPositionState
                )
            }
```

#### Stack trace
```
 java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/maps/android/SphericalUtil;
                                                                                                    	at com.google.maps.android.compose.widgets.ScaleBarKt$ScaleBar$1$1.invoke(ScaleBar.kt:284)
                                                                                                    	at com.google.maps.android.compose.widgets.ScaleBarKt$ScaleBar$1$1.invoke(ScaleBar.kt:82)
                                                                                                    	at androidx.compose.ui.draw.DrawBackgroundModifier.draw(DrawModifier.kt:104)
                                                                                                    	at androidx.compose.ui.node.BackwardsCompatNode.draw(BackwardsCompatNode.kt:381)
                                                                                                    	at androidx.compose.ui.node.LayoutNodeDrawScope.draw-x_KDEd0$ui_release(LayoutNodeDrawScope.kt:92)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:371)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
                                                                                                    	at androidx.compose.ui.node.LayoutModifierNodeCoordinator.performDraw(LayoutModifierNodeCoordinator.kt:236)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
                                                                                                    	at androidx.compose.ui.node.LayoutNode.draw$ui_release(LayoutNode.kt:840)
                                                                                                    	at androidx.compose.ui.node.InnerNodeCoordinator.performDraw(InnerNodeCoordinator.kt:151)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
                                                                                                    	at androidx.compose.ui.node.LayoutModifierNodeCoordinator.performDraw(LayoutModifierNodeCoordinator.kt:236)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
                                                                                                    	at androidx.compose.ui.node.LayoutModifierNodeCoordinator.performDraw(LayoutModifierNodeCoordinator.kt:236)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
                                                                                                    	at androidx.compose.ui.node.LayoutNode.draw$ui_release(LayoutNode.kt:840)
                                                                                                    	at androidx.compose.ui.node.InnerNodeCoordinator.performDraw(InnerNodeCoordinator.kt:151)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
                                                                                                    	at androidx.compose.ui.node.LayoutModifierNodeCoordinator.performDraw(LayoutModifierNodeCoordinator.kt:236)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
                                                                                                    	at androidx.compose.ui.node.LayoutModifierNodeCoordinator.performDraw(LayoutModifierNodeCoordinator.kt:236)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
                                                                                                    	at androidx.compose.ui.node.LayoutModifierNodeCoordinator.performDraw(LayoutModifierNodeCoordinator.kt:236)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
                                                                                                    	at androidx.compose.ui.node.LayoutNode.draw$ui_release(LayoutNode.kt:840)
                                                                                                    	at androidx.compose.ui.node.InnerNodeCoordinator.performDraw(InnerNodeCoordinator.kt:151)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
                                                                                                    	at androidx.compose.ui.node.LayoutModifierNodeCoordinator.performDraw(LayoutModifierNodeCoordinator.kt:236)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
                                                                                                    	at androidx.compose.ui.node.LayoutModifierNodeCoordinator.performDraw(LayoutModifierNodeCoordinator.kt:236)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
2023-01-03 13:56:51.296 19162-19162 AndroidRuntime          com.truescreen.app.dev               E  	at androidx.compose.ui.node.LayoutNode.draw$ui_release(LayoutNode.kt:840)
                                                                                                    	at androidx.compose.ui.node.InnerNodeCoordinator.performDraw(InnerNodeCoordinator.kt:151)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:360)
                                                                                                    	at androidx.compose.ui.node.LayoutNode.draw$ui_release(LayoutNode.kt:840)
                                                                                                    	at androidx.compose.ui.node.InnerNodeCoordinator.performDraw(InnerNodeCoordinator.kt:151)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:368)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.access$drawContainedDrawModifiers(NodeCoordinator.kt:58)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator$invoke$1.invoke(NodeCoordinator.kt:397)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator$invoke$1.invoke(NodeCoordinator.kt:396)
                                                                                                    	at androidx.compose.runtime.snapshots.Snapshot$Companion.observe(Snapshot.kt:2139)
                                                                                                    	at androidx.compose.runtime.snapshots.SnapshotStateObserver$observeReads$1$1.invoke(SnapshotStateObserver.kt:130)
                                                                                                    	at androidx.compose.runtime.snapshots.SnapshotStateObserver$observeReads$1$1.invoke(SnapshotStateObserver.kt:126)
                                                                                                    	at androidx.compose.runtime.SnapshotStateKt__DerivedStateKt.observeDerivedStateRecalculations(DerivedState.kt:341)
                                                                                                    	at androidx.compose.runtime.SnapshotStateKt.observeDerivedStateRecalculations(Unknown Source:1)
                                                                                                    	at androidx.compose.runtime.snapshots.SnapshotStateObserver.observeReads(SnapshotStateObserver.kt:126)
                                                                                                    	at androidx.compose.ui.node.OwnerSnapshotObserver.observeReads$ui_release(OwnerSnapshotObserver.kt:120)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.invoke(NodeCoordinator.kt:396)
                                                                                                    	at androidx.compose.ui.node.NodeCoordinator.invoke(NodeCoordinator.kt:58)
                                                                                                    	at androidx.compose.ui.platform.RenderNodeApi29.record(RenderNodeApi29.android.kt:180)
                                                                                                    	at androidx.compose.ui.platform.RenderNodeLayer.updateDisplayList(RenderNodeLayer.android.kt:298)
                                                                                                    	at androidx.compose.ui.platform.AndroidComposeView.dispatchDraw(AndroidComposeView.android.kt:1010)
                                                                                                    	at android.view.View.draw(View.java:23204)
                                                                                                    	at android.view.View.updateDisplayListIfDirty(View.java:22068)
                                                                                                    	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4513)
                                                                                                    	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4486)
                                                                                                    	at android.view.View.updateDisplayListIfDirty(View.java:22024)
                                                                                                    	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4513)
                                                                                                    	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4486)
                                                                                                    	at android.view.View.updateDisplayListIfDirty(View.java:22024)
                                                                                                    	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4513)
                                                                                                    	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4486)
                                                                                                    	at android.view.View.updateDisplayListIfDirty(View.java:22024)
                                                                                                    	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4513)
                                                                                                    	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4486)
                                                                                                    	at android.view.View.updateDisplayListIfDirty(View.java:22024)
                                                                                                    	at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:689)
                                                                                                    	at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:695)
                                                                                                    	at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:793)
                                                                                                    	at android.view.ViewRootImpl.draw(ViewRootImpl.java:4670)
                                                                                                    	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:4381)
                                                                                                    	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3600)
                                                                                                    	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2328)
                                                                                                    	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9080)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1231)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1239)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:899)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:832)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1214)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:942)
2023-01-03 13:56:51.296 19162-19162 AndroidRuntime          com.truescreen.app.dev               E  	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:201)
                                                                                                    	at android.os.Looper.loop(Looper.java:288)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7872)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
                                                                                                    Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.maps.android.SphericalUtil" on path: DexPathList[[dex file "/data/data/com.truescreen.app.dev/code_cache/.overlay/base.apk/classes21.dex", dex file "/data/data/com.truescreen.app.dev/code_cache/.overlay/base.apk/classes28.dex", dex file "/data/data/com.truescreen.app.dev/code_cache/.overlay/base.apk/classes16.dex", dex file "/data/data/com.truescreen.app.dev/code_cache/.overlay/base.apk/classes35.dex", zip file "/data/app/~~vWQOHFhQbB78ovc7QQN2Uw==/com.truescreen.app.dev-v0F_foDcFvuQPrkJlJCIvg==/base.apk"],nativeLibraryDirectories=[/data/app/~~vWQOHFhQbB78ovc7QQN2Uw==/com.truescreen.app.dev-v0F_foDcFvuQPrkJlJCIvg==/lib/arm64, /data/app/~~vWQOHFhQbB78ovc7QQN2Uw==/com.truescreen.app.dev-v0F_foDcFvuQPrkJlJCIvg==/base.apk!/lib/arm64-v8a, /system/lib64, /system_ext/lib64]]
                                                                                                    	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:259)
                                                                                                    	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
                                                                                                    	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
                                                                                                    	... 100 more
```
If I add the dependency `implementation("com.google.maps.android:android-maps-utils:2.3.0")` I don't get a crash.

**Comment by wangela** [comment]
@FilippoVigani Thank you for opening this issue. 🙏
Please check out these other resources that might help you get to a resolution in the meantime:

- [Support Console](https://cloud.google.com/support/) - If you have a support contract.
- [Discord](https://discord.gg/9UpV862xZ9) - chat with other developers
- [StackOverflow](http://stackoverflow.com/questions/tagged/google-maps) - use the `google-maps` tag
- [Google Maps Platform issue trackers](https://developers.google.com/maps/support/#issue_tracker)


_This is an automated message, feel free to ignore._

**Comment by DSteve595** [comment]
Thank you for the detailed info! I believe this is happening due to missing dependency info in our Maven POM, which should be fixed by #243. Will revisit this once that's merged.

**Comment by DSteve595** [comment]
@FilippoVigani This should be resolved in 2.9.1. Would greatly appreciate if you could try it out and let us know!
